File: test/java_impl_tests/src/test/java/com/nxp/erpc/tests/client/TestConst.java
/*
 * Copyright 2023 NXP
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

package com.nxp.erpc.tests.client;

import com.nxp.erpc.tests.TestingClient;
import com.nxp.erpc.tests.test_arrays.test_unit_test_common.client.CommonClient;
import com.nxp.erpc.tests.test_const.erpc_outputs.test.common.Constants;
import com.nxp.erpc.tests.test_const.erpc_outputs.test.interfaces.IEmptyInterface;
import org.junit.jupiter.api.*;

import static org.junit.jupiter.api.Assertions.*;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class TestConst extends TestingClient {
    @Test
    public void runTestConstServiceID(){
        assertEquals(2, IEmptyInterface.SERVICE_ID);
    }

    @Test
    public void runTestCheckConsts(){
        assertEquals(3, Constants.a);
        assertEquals(3.14F, Constants.b);
        assertEquals("feedbabe", Constants.c);
        assertEquals(3.14F, Constants.d);
        assertEquals(11, Constants.x);
        assertEquals(20, Constants.y);
        assertEquals(100, Constants.mass);
        assertEquals(-9.8F, Constants.accel);
    }

    @AfterAll
    public void quit() {
        new CommonClient(clientManager).quit();
    }
}

File: test/python_impl_tests/scripts/test_const/test_const.py
#!/usr/bin/python

# Copyright 2021 NXP
# All rights reserved.
#
# SPDX-License-Identifier: BSD-3-Clause

import inspect, os, sys
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))))
from server.runner import Runner
from service.test_const.erpc_outputs import test as test_const
from service.test_const.erpc_outputs import test_unit_test_common
from common.common import CommonTestServiceHandler, runTesttestQuit

###############################################################################
# Variables
###############################################################################

###############################################################################
# Functions
###############################################################################

###############################################################################
# Server
###############################################################################

## Add service handler
class ConstTestServiceHandler(test_const.interface.IEmptyInterface):
    pass

###############################################################################
# Client
###############################################################################

## Individual test cases / eRPC calls
def runTestCheckConsts(client):
    assert test_const.common.a == 3
    assert test_const.common.b == 3.14
    assert test_const.common.c == "feedbabe"
    assert test_const.common.d == 3.14
    assert test_const.common.x == 11
    assert test_const.common.y == 20
    assert test_const.common.mass == 100
    assert test_const.common.accel == -9.8

def runTestConstServiceID(client):
    assert 2 == test_const.interface.IEmptyInterface.SERVICE_ID 

###############################################################################
# PYTESTS
###############################################################################
class TestConst:
    server_cmd = [(os.path.abspath(__file__)).replace(os.sep, '/'), '-s']
    services = [test_const.server.EmptyInterfaceService(ConstTestServiceHandler()), test_unit_test_common.server.CommonService(CommonTestServiceHandler())]
    clientsConstructors = [test_const.client.EmptyInterfaceClient, test_unit_test_common.client.CommonClient]
    runner = None

    def test_runTesttestPrepareEnv(self):
        TestConst.runner = Runner('test_const', TestConst.server_cmd, TestConst.services, TestConst.clientsConstructors)

    def test_runTestConstServiceID(self):
        self.runner.runTest(test_const.interface.IEmptyInterface.SERVICE_ID, runTestConstServiceID)

    def test_runTestCheckConsts(self):
        self.runner.runTest(test_const.interface.IEmptyInterface.SERVICE_ID, runTestCheckConsts)

    def test_runTesttestQuit(self):
        self.runner.runTest(test_unit_test_common.interface.ICommon.SERVICE_ID, runTesttestQuit)

    def runTests(self):
        self.test_runTesttestPrepareEnv()
        self.test_runTesttestQuit()
        self.test_runTestConstServiceID()
        self.test_runTestCheckConsts()

###############################################################################
# Main
###############################################################################

if __name__ == "__main__":
    tester = TestConst()
    tester.runTests()
